src/js/admin/index.js

```diff
@@ -0,0 +1,20 @@
+document.querySelector( '.media-single-image button.browse' ).addEventListener( 'click', selectImage );
+
+function selectImage() {
+	const optionName = this.dataset.option;
+
+	const imageFrame = wp.media( {
+		multiple: false,
+		library: {
+			type: 'image',
+		},
+	} );
+
+	imageFrame.on( 'select', function() {
+		const url = imageFrame.state().get( 'selection' ).first().toJSON().url;
```

Comment:
I would be a bit more defensive in here. Every call here is nullable, so could we have some kind of checks to ensure that nothing fails? If something fails, I would display an `alert`.